test/server/cards/01_SOR/units/LeiaOrganaDefiantPrincess.spec.js

```diff
@@ -0,0 +1,44 @@
+describe('Leia Organa, Defiant Princess', function() {
+    integration(function() {
+        describe('Leia Organa, Defiant Princess\'s when played ability', function() {
+            beforeEach(function () {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['leia-organa#defiant-princess'],
+                        groundArena: [{ card: 'atst', exhausted: true }],
+                        spaceArena: [{ card: 'cartel-spacer', exhausted: false }],
+                        leader: { card: 'jyn-erso#resisting-oppression', deployed: true, exhausted: false }
+                    },
+                    player2: {
+                        groundArena: [{ card: 'wampa', exhausted: false }],
+                        spaceArena: [{ card: 'alliance-xwing', exhausted: true }]
+                    }
+                });
+            });
+
+            it('should be able to ready a friendly resource', function () {
+                this.player1.clickCard(this.leiaOrgana);
+                expect(this.player1).toHaveEnabledPromptButtons(['Ready a resource', 'Exhaust a unit']);
+                const exhaustedResourcesBeforeAbility = this.player1.countExhaustedResources();
+                this.player1.clickPrompt('Ready a resource');
+                expect(this.player1.countExhaustedResources()).toBe(exhaustedResourcesBeforeAbility - 1);
+            });
+
+            it('should be able to exhaust an enemy unit', function () {
+                this.player1.clickCard(this.leiaOrgana);
+                this.player1.clickPrompt('Exhaust a unit');
+                expect(this.player1).toBeAbleToSelectExactly([this.leiaOrgana, this.atst, this.cartelSpacer, this.jynErso, this.wampa, this.allianceXwing]);
+                this.player1.clickCard(this.wampa);
+                expect(this.wampa.exhausted).toBe(true);
+            });
+
+            it('should be able to exhaust a friendly unit', function () {
```

Comment:
I think Chaqa wrote the note in the excel that we need to test exhausting both a friendly and an enemy unit, this test feels unnecessary to me with all the other tests we have in place so I'd prefer to cut it.

I would like though to test what happens if there is no other unit on the field and the player selects "exhaust a unit," can we confirm that nothing happens? Since basically it would auto-target Leia as the only unit and she's exhausted already. But the button should still appear correctly.